util/stopper_test.go

```diff
@@ -237,3 +236,52 @@ func TestStopperClosers(t *testing.T) {
 		t.Errorf("expected true & true; got %t & %t", tc1, tc2)
 	}
 }
+
+func TestStopperNumTasks(t *testing.T) {
+	s := NewStopper()
+
+	var tasks []PendingTask
+	for i := 0; i < 3; i++ {
+		tasks = append(tasks, s.StartTask())
+		if numTypes, numTasks := len(s.tasks), s.NumTasks(); numTypes != 1 || numTasks != i+1 {
+			t.Errorf("stopper should have %d running tasks, got %d / %+v", i+1, s.NumTasks(), s.tasks)
+		}
+		m := s.RunningTasks()
+		if len(m) != 1 {
+			t.Fatalf("expected exactly one task map entry: %+v", m)
```

Comment:
I prefer to use `t.Fatal` by default and only use `t.Error` when there is a reasonable expectation that collecting multiple errors will be informative. The big reason is that if there is a panic (which is fairly likely if the error resulted in some pointer being set to nil), the test process is simply killed (with a stack trace at the point of the panic) and the errors that have been collected by `t.Error` are never printed.